core/src/gl/texture.cpp

```diff
@@ -73,120 +63,100 @@ bool Texture::loadImageFromMemory(const std::vector<char>& _data) {
     // texture data but a Tangram style shader requires a shader sampler
     GLuint blackPixel = 0x0000ff;
 
-    setData(&blackPixel, 1);
+    setPixelData(&blackPixel, 1);
 
     return false;
 }
 
-Texture::Texture(Texture&& _other) {
+Texture::Texture(Texture&& _other) noexcept {
     *this = std::move(_other);
 }
 
-Texture& Texture::operator=(Texture&& _other) {
+Texture& Texture::operator=(Texture&& _other) noexcept {
     m_glHandle = _other.m_glHandle;
     _other.m_glHandle = 0;
 
     m_options = _other.m_options;
     m_data = std::move(_other.m_data);
-    m_dirtyRanges = std::move(_other.m_dirtyRanges);
+    m_dirtyRows = std::move(_other.m_dirtyRows);
     m_shouldResize = _other.m_shouldResize;
     m_width = _other.m_width;
     m_height = _other.m_height;
-    m_target = _other.m_target;
-    m_generateMipmaps = _other.m_generateMipmaps;
     m_rs = _other.m_rs;
 
     return *this;
 }
 
-void Texture::setData(const GLuint* _data, unsigned int _dataSize) {
+void Texture::setPixelData(const GLuint *data, size_t length) {
```

Comment:
nit: styling, `GLuint*` instead of `GLuint *`

Reply:
Forgot to update my CLion style settings haha